### ISSUE TITLE
segfault for ABNF range production

### ISSUE THREAD

**Issue by reschke** [katef/kgt#19]
The following ABNF rule:

~~~
obs-text = %x80-FF
~~~

causes a segmentation fault (Ubuntu-on-Windows).

**Comment by katef** [comment]
Feels like `unsigned char` somewhere:
```
; echo -e 'a = %x00-7F' | ./build/bin/kgt -l abnf -e rrtext 
a:
    ||--v-- "\x00" -->--||
        |            |
        |     :      |
        |            |
        `-- "\x7f" --'

; echo -e 'a = %x00-80' | ./build/bin/kgt -l abnf -e rrtext
a:
malloc: Cannot allocate memory
; 
```

**Comment by katef** [comment]
```
; echo -e 'a = %x00-80' | ./build/bin/kgt -l abnf -e rrtdump
a: (before prettify)
    VLIST w=12 a=0 d=4294967295 o=0: [
        HLIST w=6 a=0 d=4294967295: [
            VLIST w=6 a=0 d=4294967295 o=0: [
            ]
        ]
    ]
```

**Comment by katef** [comment]
The `%x80` case fixed here: https://github.com/katef/kgt/commit/545aeb93071a59a9820ebde743d9022037bb17d0
That still leaves the `%xFF` case, which probably overflows to 0.

**Comment by katef** [comment]
```
; echo -e 'a = %xFE-FF' | ./build/bin/kgt -l abnf -e rrtdump
a: (before prettify)
    VLIST w=18 a=0 d=3 o=0: [
        HLIST w=12 a=0 d=3: [
            VLIST w=12 a=0 d=3 o=0: [
                HLIST w=6 a=0 d=1: [
zsh: done                echo -e 'a = %xFE-FF' | 
zsh: segmentation fault  ./build/bin/kgt -l abnf -e rrtdump
```

**Comment by katef** [comment]
Overflow at end of range fixed here: https://github.com/katef/kgt/pull/21/commits/505bf5b1d246fc612235deb994cfe64e8c4ff747

**Comment by katef** [comment]
Merged in #21
